tee-worker/litentry/core/assertion-build/src/dynamic/scripts/deploy.ts

```diff
@@ -0,0 +1,322 @@
+                                    console.log(
+                                        `\n${event.section}.${event.method}, ${blockHash}`
+                                    )
+                                    if (
+                                        event.section === 'evmAssertions' &&
+                                        event.method === 'AssertionStored'
+                                    ) {
+                                        deploySuccess = true
+                                    }
+                                }
+                            }
+                        }
+                    )
+
+                if (!deploySuccess) {
+                    blocksToScan -= 1
+                    if (blocksToScan < 1) {
+                        console.log(
+                            '\nTimed out listening for deploy result event'
+                        )
+                        resolve({
+                            success: deploySuccess,
+                            hashes: relatedHashes,
+                        })
+                        ;(await unsubscribe)()
+                    }
+                    return
+                }
+
+                resolve({ success: deploySuccess, hashes: relatedHashes })
+                ;(await unsubscribe)()
+            }
+        )
+    })
+}
+
+/**
+ * Generates a contract ID based on the provided contract byte code and secrets.
+ *
+ * Same contract byte code and secrets will get the same contractId, this can ensure dev/stg/prod has same contract ID.
+ *
+ * @param {string} contractByteCode - The byte codes of the contract.
+ * @param {string[]} secrets - An array of secret strings to be included in the hash.
+ * @returns {string} The generated contract ID.
+ */
+function generateContractId(
+    contractByteCode: string,
+    secrets: string[]
+): string {
+    // Combine the contract byte codes and secrets into a single string
+    const data = contractByteCode + secrets.join(' ')
+
+    // Create a SHA-256 hash of the combined data
+    const hash = crypto.createHash('sha256').update(data).digest('hex')
+
+    // Take the first 40 characters of the hash to form the contract ID, prefixed with '0x'
+    const contractId = `0x${hash.slice(0, 40)}`
+    return contractId
+}
+
+function genesisSubstrateWallet(name: string) {
+    const keyring = new Keyring({ type: 'sr25519' })
+    const keyPair = keyring.addFromUri(`//${name}`, { name })
+    return keyPair
+}
+
+/**
+ * Encrypts a plaintext buffer using the provided public key in segments.
+ *
+ * Same logic as: https://github.com/apache/incubator-teaclave-sgx-sdk/blob/master/sgx_crypto_helper/src/rsa3072.rs#L161-L179
+ *
+ * @param {crypto.KeyLike} pubKey - The public key to use for encryption.
+ * @param {Uint8Array} plaintext - The plaintext buffer to encrypt.
+ * @returns {Buffer} The encrypted data.
+ */
+function encryptBuffer(pubKey: crypto.KeyLike, plaintext: Uint8Array): Buffer {
```

Comment:
Good spot, shall we use it ubiquitously in our SDK?

cc @jonalvarezz too

Reply:
Yes, we need to do segmented encryption for all over limited request payload.